### PR TITLE
Refactoring Code based on Design and Implementation Smells

### PR DESCRIPTION
**Pull request by nisarrg**

### CODE REVIEW

**Comment by nisarrg** [comment]
Fixing Code Smells in the code.

File: technology/java-spring-boot/src/main/java/org/acme/schooltimetabling/solver/TimeTableConstraintProvider.java [hunk 1]
```diff
@@ -12,9 +12,12 @@
 
 public class TimeTableConstraintProvider implements ConstraintProvider {
 
+    private static final int MAX_GAP_MINUTES = 30;
+
+    //TODO --> tool detected this as long statement, it is technically not : FP
```
Comment (triceo) [inline technology/java-spring-boot/src/main/java/org/acme/schooltimetabling/solver/TimeTableConstraintProvider.java:17]: Please remove the comment.

File: technology/java-spring-boot/src/main/java/org/acme/schooltimetabling/solver/TimeTableConstraintProvider.java [hunk 2]
```diff
@@ -70,21 +73,23 @@ Constraint teacherRoomStability(ConstraintFactory constraintFactory) {
                 .asConstraint("Teacher room stability");
     }
 
-    Constraint teacherTimeEfficiency(ConstraintFactory constraintFactory) {
-        // A teacher prefers to teach sequential lessons and dislikes gaps between lessons.
+
+    //TODO ---> removed magic number and introduced global variable : TP
```
Comment (triceo) [inline technology/java-spring-boot/src/main/java/org/acme/schooltimetabling/solver/TimeTableConstraintProvider.java:77]: Please remove the comment.

File: technology/java-spring-boot/src/main/java/org/acme/schooltimetabling/solver/TimeTableConstraintProvider.java [hunk 3]
```diff
@@ -70,21 +73,23 @@ Constraint teacherRoomStability(ConstraintFactory constraintFactory) {
...
                     Duration between = Duration.between(lesson1.getTimeslot().getEndTime(),
                             lesson2.getTimeslot().getStartTime());
-                    return !between.isNegative() && between.compareTo(Duration.ofMinutes(30)) <= 0;
+                    return !between.isNegative() && between.compareTo(Duration.ofMinutes(MAX_GAP_MINUTES)) <= 0;
                 })
                 .reward(HardSoftScore.ONE_SOFT)
                 .asConstraint("Teacher time efficiency");
     }
 
+    //TODO ---> removed magic number and introduced global variable : TP
```
Comment (triceo) [inline technology/java-spring-boot/src/main/java/org/acme/schooltimetabling/solver/TimeTableConstraintProvider.java:92]: Please remove the comment.

File: use-cases/maintenance-scheduling/src/main/java/org/acme/maintenancescheduling/domain/MaintenanceSchedule.java [hunk 4]
```diff
@@ -31,6 +31,7 @@ public class MaintenanceSchedule {
     // Ignored by Timefold, used by the UI to display solve or stop solving button
     private SolverStatus solverStatus;
 
+    //TODO --> Never used Constructor.
```
Comment (triceo) [inline use-cases/maintenance-scheduling/src/main/java/org/acme/maintenancescheduling/domain/MaintenanceSchedule.java:34]: Please remove the comment. The constructor is required for Timefold to function, even though it seems unused.

File: use-cases/maintenance-scheduling/src/main/java/org/acme/maintenancescheduling/domain/MaintenanceSchedule.java [hunk 5]
```diff
@@ -42,16 +43,17 @@ public MaintenanceSchedule(WorkCalendar workCalendar,
...
+//    @ValueRangeProvider
+//    public List<LocalDate> createStartDateList() {
+//        return workCalendar.getFromDate().datesUntil(workCalendar.getToDate())
+//                // Skip weekends. Does not work for holidays.
+//                // Keep in sync with EndDateUpdatingVariableListener.updateEndDate().
+//                // To skip holidays too, cache all working days in WorkCalendar.
+//                .filter(date -> date.getDayOfWeek() != DayOfWeek.SATURDAY
+//                        && date.getDayOfWeek() != DayOfWeek.SUNDAY)
+//                .collect(Collectors.toList());
+//    }
```
Comment (triceo) [inline use-cases/maintenance-scheduling/src/main/java/org/acme/maintenancescheduling/domain/MaintenanceSchedule.java:56]: Why was this removed? I do not see it being replaced by anything.

File: use-cases/vehicle-routing-capacity/src/main/java/org/acme/vehiclerouting/rest/VehicleRouteDemoResource.java [hunk 6]
```diff
@@ -35,14 +35,14 @@ public enum DemoData {
...
-        private Location southWestCorner;
-        private Location northEastCorner;
+        final private int customerCount;
+        final private int vehicleCount;
+        final private int depotCount;
+        final private int minDemand;
+        final private int maxDemand;
+        final private int vehicleCapacity;
+        final private Location southWestCorner;
+        final private Location northEastCorner;
```
Comment (triceo) [inline use-cases/vehicle-routing-capacity/src/main/java/org/acme/vehiclerouting/rest/VehicleRouteDemoResource.java:45]: Please use `private final`, not `final private`.

File: use-cases/vehicle-routing-time-windows/src/main/java/org/acme/vehiclerouting/rest/VehicleRouteDemoResource.java [hunk 7]
```diff
@@ -48,41 +48,24 @@ public enum DemoData {
...
         FIRENZE(77, 6, 2, LocalTime.of(7, 30),
                 new Location(43.751466, 11.177210), new Location(43.809291, 11.290195));
 
-        private int customerCount;
-        private int vehicleCount;
-        private int depotCount;
-        private LocalTime vehicleStartTime;
-        private Location southWestCorner;
-        private Location northEastCorner;
+        // TODO --> these variables will be final : FN
```
Comment (triceo) [inline use-cases/vehicle-routing-time-windows/src/main/java/org/acme/vehiclerouting/rest/VehicleRouteDemoResource.java:57]: Please remove the comment.

**Review by triceo — CHANGES_REQUESTED** [review]
Thank you for the PR, @nisarrg!
I am requesting some changes to be made before we can merge it.

File: technology/java-spring-boot/src/main/java/org/acme/schooltimetabling/TimeTableSpringBootApp.java [hunk 8]
```diff
@@ -16,6 +16,7 @@
 import org.springframework.context.annotation.Bean;
 import org.springframework.data.domain.Sort;
 
+// TODO --> unutilised abstraction --> main file : FP
```
Comment (triceo) [inline technology/java-spring-boot/src/main/java/org/acme/schooltimetabling/TimeTableSpringBootApp.java:19]: Please remove the comment.

File: technology/java-spring-boot/src/main/java/org/acme/schooltimetabling/persistence/TimeTableRepository.java [hunk 9]
```diff
@@ -11,6 +11,7 @@
 @Transactional
 public class TimeTableRepository {
 
+    // TODO --> SINGLETON_TIME_TABLE_ID is public : deficient encapsulation --> TP
```
Comment (triceo) [inline technology/java-spring-boot/src/main/java/org/acme/schooltimetabling/persistence/TimeTableRepository.java:14]: Please remove the comment.

**Comment by triceo** [comment]
Also, please open the PR against the `development` branch. We do not merge any changes to `stable` branch.

**Comment by triceo** [comment]
Closing on account of no response.
Please reopen if you intend to keep working on it.

**Pull request closed without merging**